doc/source/cluster/kubernetes/user-guides/kuberay-auth.md

```diff
@@ -0,0 +1,200 @@
+
+## Verify Initial Unauthorized Access
+
+Attempt to submit a Ray job to the cluster to verify that authentication is required. You should receive a `401 Unauthorized` error:
+
+```bash
+kubectl port-forward svc/ray-cluster-with-auth-head-svc 8265:8265 &
+ray job submit --address http://localhost:8265  -- python -c "import ray; ray.init(); print(ray.cluster_resources())"
+```
+
+You'll see an error similar to this:
+
+```
+...
+requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: http://localhost:8265/api/version
+```
+
+This confirms that the Ray cluster requires authentication.
+
+## Configure Kubernetes RBAC for Access Control
+
+To access the RayCluster, you need:
+*  **Authentication:** Provide a valid authentication token (e.g., a Kubernetes service account token or a cloud IAM token) in the request headers.
+*  **Authorization:** Your authenticated user or service account must have the necessary Kubernetes RBAC permissions to access the `RayCluster` resource.
+
+This guide demonstrates granting access using a Kubernetes service account, but the same principles apply to individual Kubernetes users or cloud IAM users.
+
+### Create a Kubernetes Service Account
+
+Create a service account that will represent your Ray job submitter:
+
+```bash
+kubectl create serviceaccount ray-user
+```
+
+Confirm that the service account currently cannot access the `RayCluster` resource:
+
+```bash
+kubectl auth can-i get rayclusters.ray.io/ray-cluster-with-auth --as=system:serviceaccount:default:ray-user
+```
+
+The output should be `no`.
+
+### Grant Access using Kubernetes RBAC
+
+Create a `Role` and `RoleBinding` to grant the necessary permissions to the `ray-user` service account:
+
+```yaml
+# ray-cluster-rbac.yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: ray-user
+  namespace: default
+rules:
+- apiGroups: ["ray.io"]
+  resources:
+  - 'rayclusters'
+  verbs: ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: ray-user
+  namespace: default
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: ray-user
+subjects:
+- kind: ServiceAccount
+  name: ray-user
+  namespace: default
+```
+
+Apply the RBAC configuration:
+
+```bash
+kubectl apply -f ray-cluster-rbac.yaml
+```
+
+### Verify Access
+
+Confirm that the service account now has access to the `RayCluster` resource:
+
+```bash
+kubectl auth can-i get rayclusters.ray.io/ray-cluster-with-auth --as=system:serviceaccount:default:ray-user
+```
+
+The output should be `yes`.
+
+## Submit a Ray Job with Authentication
+
+Now you can submit a Ray job using the service account's authentication token.
+
+Get a token for the `ray-user` service account and store it in the `RAY_JOB_HEADERS` environment variable:
+
+```bash
+export RAY_JOB_HEADERS="{\"Authorization\": \"Bearer $(kubectl create token ray-user)\"}"
```

Comment:
Can you add a note that kubectl create token is only supported in kubectl versions >= 1.24 ([ref](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md)) or update the "Prerequisites" section?

<img width="945" alt="image" src="https://github.com/user-attachments/assets/2a6e7fd4-12a4-48f1-b86f-f5f1047d1de4">

Reply:
1.24 has been EOL for a while now, it's probably safe to assume most users are using a newer version at this point

Reply:
I followed this documentation and realized that I am using version 1.23.X on the client side. On the server side, I think most people use newer Kubernetes versions, but on the client side, users might not upgrade as frequently.

Reply:
good point, I'll add a note about this

Reply:
done